catalog/core/catalog-core-api/src/main/java/ddf/catalog/operation/Request.java

```diff
@@ -13,12 +13,47 @@
  */
 package ddf.catalog.operation;
 
+import java.util.HashSet;
+import java.util.Set;
+
+import ddf.catalog.Constants;
+
 /**
  * Marker interface to differentiate {@link Request}s from {@link Response}s.
  *
  * @author michael.menousek@lmco.com
- *
  */
 public interface Request extends Operation {
 
+    /**
+     * Returns a set of ids that correspond to catalog destinations
+     * @return
+     */
+    default Set<String> getDestinationIds() {
+        return new HashSet<>();
+    }
+
+    /**
+     * Returns true if this request will be run on a remote catalog. This does not mean that
+     * this same request will not be run on the local catalog. Default implementation requires
+     * the catalog framework to set the needed properties for this method to work.
+     *
+     * @return
+     */
+    default boolean isRemoteRequest() {
```

Comment:
It might be better to move convenience methods to an implementation instead of adding default methods to the interface.